### PR TITLE
Feature block content edit view [View]

### LINKED ISSUE #343: [PE-D] Difficult to edit notes
The current 'edit' command replaces the entire contents in a note.

As Notably is a desktop app for managing notes, it might be better if there are more commands to edit a pre existing note.
Some suggestions are: 
- a separate command to edit the header and body
- a separate command to add an additional line of text to a pre existing note
- you might also want to allow users to rename the name of a note

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: beatricetay/ped#7

### LINKED ISSUE #360: [PE-D] Unable to have linebreak in body section of the note
In the UG section 3.3, it was mentioned that newline could be used in the body paragraph. However, pressing the enter button for new line will just execute the commnad. I tried using \n and "\n" and '\n' and it does not work as well. 

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: CheyanneSim/ped#3

### LINKED ISSUE #363: [PE-D] Editing a note is equivalent to rewriting the whole note
As of now, editing a note means that the user is rewriting the entire note. This feels quite impractical if a user simply wants to change a few minor details in the note's body, and save the changes. Right now, if the user wants to do so, they have to copy the note's original body, make their changes in the app CLI/externally on their desktop, and then create a brand new note with the updated body. This makes it quite inconvenient for a typical user.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: chishanw/ped#6

### LINKED ISSUE #366: [PE-D] Incorrect parsing for edit command
From the User Guide, the edit command is `edit -b BODY`.
However, I have discovered that simply typing `edit BODY` works as well.
Another edge case that you might want to consider is if the user types `edit -t BODY`, because of the above issue, the body of the note becomes `-t BODY`, but maybe an error should be thrown to let the user know that the title of the note cannot be edited.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: chishanw/ped#3

### LINKED ISSUE #374: [PE-D] Not able to add in text in the body
The app only allow us to replace the body content with a totally new one by using edit command, i think it will be much better for the user to be able to update the body based on the original version.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: Kangwkk/ped#4

### PR DESCRIPTION
**Pull request by HemanshuGandhi**
Part of Hemanshu's deliverables as in #380.

- Adds a Block Content Edit View that can be opened via the "edit" command.
- Block Content Edit View must be closed using the "ESC" key.
- On closure, the new contents are passed to `Logic`, which then updates the `Block`'s contents. (Autosave behavior).

### CODE REVIEW

File: src/main/java/com/notably/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -28,7 +28,8 @@ public EditCommand(Body body) {
     public void execute(Model notablyModel) throws CommandException {
         requireNonNull(notablyModel);
         try {
-            notablyModel.updateCurrentlyOpenBlockBody(body);
+            //notablyModel.updateCurrentlyOpenBlockBody(body);
```
Comment (HemanshuGandhi) [inline src/main/java/com/notably/logic/commands/EditCommand.java]: @ljiazh3ng just take note of this, as per our discussion.

File: src/main/java/com/notably/view/MainWindow.java [hunk 2]
```diff
@@ -79,6 +81,24 @@ public Stage getPrimaryStage() {
...
+            if (isFocused) {
+                mainWindow.setEffect(null);
+            } else {
+                ColorAdjust colorAdjust = new ColorAdjust();
+                colorAdjust.setBrightness(-0.4);
+                mainWindow.setEffect(colorAdjust);
+            }
+        }));
+    }
+
```
Comment (kevinputera) [inline src/main/java/com/notably/view/MainWindow.java:101]: A consequence of this listener is that the whole Notably window got darkened when other apps are focused. What do you think about this? Should we just leave it as is for now?

File: src/main/java/com/notably/view/blockcontent/BlockContentEditView.java [hunk 3]
```diff
@@ -1,63 +1,253 @@
...
+     * Triggers changes in the visibility of the Edit modal when certain Keystrokes are registered.
+     * Primarily, allows for the user to exit the Block Edit modal using the "ESC" keyboard shortcut.
+     */
+    private void setKeyboardListeners() {
+        stage.addEventHandler(KeyEvent.KEY_RELEASED, (KeyEvent event) -> {
+            if (KeyCode.ESCAPE == event.getCode()) {
+                handleClose();
+            }
+        });
+    }
```
Comment (kevinputera) [inline src/main/java/com/notably/view/blockcontent/BlockContentEditView.java]: Do you think we should add another shortcut that is more relevant to the general user? Something like `ctrl/cmd + s`

**Review by kevinputera — APPROVED** [review]
LGTM, @HemanshuGandhi are you going to add the title info and path info header to `BlockContentDisplayView`?

File: src/main/java/com/notably/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -28,7 +28,8 @@ public EditCommand(Body body) {
...
+            notablyModel.setBlockEditable(true);
         } catch (CannotModifyRootException ex) {
             throw new CommandException(ex.getMessage());
         }
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/commands/EditCommand.java]: @HemanshuGandhi you can actually edit this directly to

```java
public void execute(Model notablyModel) {
  requireNonNull(notablyModel);
  notablyModel.setBlockEditable(true);
}
```

**Review by kevinputera — COMMENTED** [review]
Some more comments!

File: src/main/java/com/notably/logic/commands/EditCommand.java (hunk 4 above)
Comment (HemanshuGandhi) [reply src/main/java/com/notably/logic/commands/EditCommand.java]: alright! Rectified.

File: src/main/java/com/notably/view/blockcontent/BlockContentEditView.java (hunk 3 above)
Comment (HemanshuGandhi) [reply src/main/java/com/notably/view/blockcontent/BlockContentEditView.java]: Good Suggestion! `ctrl/cmd + s` might indeed be more intuitive. I will set it up such that the same behavior on pressing `ESC` happens with `ctrl/cmd + s`.

File: src/main/java/com/notably/view/MainWindow.java (hunk 2 above)
Comment (HemanshuGandhi) [reply src/main/java/com/notably/view/MainWindow.java:101]: Oh, I see. Could you attach a screenshot of this behavior? I have not noticed this on my mac!

**Pull request merged**